### PR TITLE
Drop python2 support in travis - #544

### PR DESCRIPTION
**Pull request by swamper123**

### CODE REVIEW

**Comment by swamper123** [comment]
This shall fix and do:
- fixed OSX test fails
- Drop Python 2
- Add Python 3.9 support
- Drop Ubuntu xenial (16.04)
- Upgrades Ubuntu up to 18.04 (bionic)
- Drops xcode 8.3 (deprecations denied test start)
- Upgrade to xcode12.2 tests

**Comment by dhoomakethu** [comment]
Python2 support will be removed from Pymodbus 3.x onwards.

**Review by dhoomakethu — APPROVED** [review]

**Comment by swamper123** [comment]
@dhoomakethu I want to mention that [a PR is ready, based on this PR, to get rid of Python2 code and < Python 3.6](https://github.com/swamper123/pymodbus/tree/remove_python2_compatibility_code).
I chose the 3.6+ support to get rid of dead versions and also the async stuff is "nearly" the same since then.

Now I have no clue how this works here on github...how this may be marked as goal or something like that .... it's my first time that such a thing has happened. 😄 I just want to let you know that.

File: tox.ini [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # directory.
 
 [tox]
-envlist = py27, py35, py36, py37, pypy
```
Comment (JamesJeffryes) [inline tox.ini:7]: Shouldn't py39 be added to match the Travis tests?

**Comment by JamesJeffryes** [comment]
@dhoomakethu Would love to see this merged as this should fix CI errors for a number of outstanding PRs

**Comment by dhoomakethu** [comment]
@swamper123 Thanks for the fix . I will create a new 3.x base branch and merge this against it this week. I may ask you to reraise it against the 3.x branch or will cherry pick the commits from this branch (jfyi). @JamesJeffryes Will try to rollout a new version asap (no timelines guaranteed though) , expect RC in github in sometime.

File: tox.ini (hunk 1 above)
Comment (swamper123) [reply tox.ini:7]: @JamesJeffryes Sometimes people haven't wanted me to add the newest stable versions in Python. But I can add that.

**Comment by swamper123** [comment]
@dhoomakethu No worries, I will do that in the next days (perhaps even today).

**Comment by dhoomakethu** [comment]
@swamper123  I believe this has been merged in to 3.x branch. I will be closing this.

**Pull request closed without merging**